tests/conftest.py

```diff
@@ -7,19 +7,37 @@
 from pyspark.sql import SparkSession
 
 
-@pytest.fixture(scope="session")
-def spark():
-    spark = (
+_spark_session = None
+_spark_context = None
+
+
+def pytest_configure(config):
+    global _spark_session, _spark_context
+
+    _spark_session = (
         SparkSession.builder.master("local").appName("mozanalysis_test").getOrCreate()
     )
+    _spark_context = _spark_session.sparkContext
 
     logger = logging.getLogger("py4j")
     logger.setLevel(logging.ERROR)
 
-    yield spark
-    spark.stop()
+
+def pytest_unconfigure(config):
+    global _spark_session
```

Comment:
The `global` here isn't necessary since we aren't trying to assign to the name; same for the similar declarations in `spark()` and `spark_context()` -- I don't mind them if you think they're useful as signposting

Reply:
Good catch. This file went through so many variations while I was testing what would work with pytest. I'll update and merge. Thanks!